src/main/java/com/company/managementservice/controller/OrganisationController.java

```diff
@@ -0,0 +1,65 @@
+package com.company.managementservice.controller;
+
+import com.company.managementservice.exception.EmptyBodyException;
+import com.company.managementservice.exception.NotFoundException;
+import com.company.managementservice.model.dto.OrganisationDto;
+import com.company.managementservice.model.response.BaseMessageResponse;
+import com.company.managementservice.model.response.ServiceResponse;
+import com.company.managementservice.service.OrganisationService;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@Log4j2
+@RequestMapping(value = "/organisation")
+public class OrganisationController {
+
+    @Autowired
+    OrganisationService organisationService;
+
+    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> postOrganisationDetails(
+            @RequestBody OrganisationDto organisationDto) throws EmptyBodyException {
+        if (organisationDto.getName() == null || organisationDto.getHeadOfficeLocation() == null)
```

Comment:
```suggestion
        if (organisationDto.getName() == null || organisationDto.getHeadOfficeLocation() == null)
```
You can put @NonNull above variable in DTO itself.